src/components/CCPDashboard/PatientDetailsDialog.tsx

```diff
@@ -34,6 +50,75 @@ interface PatientDetail {
   styles?: object;
 }
 
+export const PatientDetailsDialogRouting = ({
+  match: {
+    params: { eventId, patientId, ccpId },
+  },
+}: {
+  match: {
+    params: {
+      eventId: string;
+      patientId?: string;
+      ccpId: string;
+    };
+  };
+}) => {
+  const [openDetails, setOpenDetails] = React.useState(true);
+  const history = useHistory();
+  const classes = useStyles();
+
+  const { data } = useQuery(
+    patientId ? GET_PATIENT_BY_ID(patientId) : GET_ALL_PATIENTS
+  );
+
+  const patient: Patient = data ? data.patient : undefined;
+
+  const handleCloseDetails = () => {
+    setOpenDetails(false);
+    history.push(`/events/${eventId}/ccps/${ccpId}`);
+  };
+
+  return (
+    <div>
+      {patient ? (
+        <Dialog
+          open={openDetails}
+          onClose={handleCloseDetails}
+          PaperProps={{ className: classes.detailsDialog }}
+        >
+          <PatientDetailsDialog
+            patient={patient}
+            onClose={handleCloseDetails}
+          />
+          <DialogActions
+            style={{
+              borderLeft: `16px solid ${
+                Colours[`Triage${capitalize(patient.triageLevel)}`]
+              }`,
+            }}
+          >
+            <Button
+              onClick={() => {
+                history.push(
+                  `/events/${eventId}/ccps/${ccpId}/patients/${patientId}`
+                );
+              }}
+              color="secondary"
+              className={classes.editButton}
+            >
+              Edit
+            </Button>
+          </DialogActions>
+        </Dialog>
+      ) : (
+        <Redirect
+          to={`/events/${eventId}/ccps/${ccpId}/patients/${patientId}`}
+        />
```

Comment:
how does this redirect work? it only does this if `patient` is falsy (which imo implies that the `patientId` is invalid), and attempts to go to the edit form for that patient (which wouldn't work if the `patientId` is invalid)

Reply:
made it redirect to /events